pkg/reconciler/app/resources/servicebinding.go

```diff
@@ -0,0 +1,97 @@
+// Copyright 2019 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the License);
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an AS IS BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package resources
+
+import (
+	"encoding/json"
+	"fmt"
+
+	"github.com/google/kf/pkg/apis/kf/v1alpha1"
+	"github.com/knative/serving/pkg/resources"
+	servicecatalogv1beta1 "github.com/poy/service-catalog/pkg/apis/servicecatalog/v1beta1"
+	servicecatalog "github.com/poy/service-catalog/pkg/svcat/service-catalog"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/labels"
+	"k8s.io/apimachinery/pkg/selection"
+	"knative.dev/pkg/kmeta"
+)
+
+const (
+	// AppNameLabel is the label used on bindings to define which app the binding belongs to.
+	AppNameLabel = "kf-app-name"
+	// BindingNameLabel is the label used on bindings to define what VCAP name the secret should be rooted under.
+	BindingNameLabel = "kf-binding-name"
+)
+
+// MakeServiceBindingLabels creates labels that can be used to tie a source to a build.
+func MakeServiceBindingLabels(app *v1alpha1.App, bindingName string) map[string]string {
+	return resources.UnionMaps(app.ComponentLabels("servicebinding"), map[string]string{
+		AppNameLabel:     app.Name,
+		BindingNameLabel: bindingName,
+	})
+}
+
+func MakeServiceBindingName(app *v1alpha1.App, binding *v1alpha1.AppSpecServiceBinding) string {
+	return fmt.Sprintf("kf-binding-%s-%s", app.Name, binding.InstanceRef.Name)
```

Comment:
Is it possible/should it be possible to bind to the same instance twice with different binding names?

Reply:
My intention was to not make this possible, based on what the code allowed before my changes. Should we allow multiple bindings from 1 app to the same service?

Reply:
The old code did not allow it: only 1 instance binding per app, instance.
But, should we allow multiple bindings? We could perform uniqueness on binding names, and format the binding name with the...binding name

Reply:
Per our discussion, we're going to allow multiple bindings